lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
I spent some time figuring out when this happens; it seems the first time you run the app and the db file does not exist, it does create the file and it saves data in your file as expected; but if you stop the server and run it, it sometimes does not write in the file. Please let me do more investigation on this to make sure...